web-api/src/main/java/dk/dbc/opensearch/input/SearchRequest.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2018 DBC A/S (http://dbc.dk/)
+ *
+ * This is part of opensearch-solr
+ *
+ * opensearch-solr is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * opensearch-solr is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package dk.dbc.opensearch.input;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import javax.xml.stream.Location;
+import javax.xml.stream.XMLStreamException;
+
+import static dk.dbc.opensearch.input.RequestHelpers.*;
+
+/**
+ *
+ * @author DBC {@literal <dbc.dk>}
+ */
+public class SearchRequest extends CommonRequest {
+
+    private static final Function<String, CollectionType> COLLECTION_TYPES = mapTo(makeTrimOneOf("collectionType", "work", "work-1", "manifestation"),
+                                                                                   CollectionType::from);
+    private static final Function<String, String> QUERY_LANGUAGES = makeTrimOneOf("queryLanguage", "cqleng", "bestMatch");
+    private static final Function<String, String> OBJECT_FORMATS = makeTrimOneOf("objectFormat",
+                                                                                 "dkabm", "docbook", "marcxchange", "opensearchobject", "briefWorkDisplay",
+                                                                                 "bibliotekdkWorkDisplay", "briefDisplayHtml", "fullDisplayHtml",
+                                                                                 "workDisplayHtml", "briefDisplay", "fullDisplay", "refWorks", "ris");
+
+    private String query = null;
+    private String queryLanguage = null;
+    private Boolean allObjects = null;
+    private CollectionType collectionType;
+    private List<Facets> facets = null;
+    private Integer collapseHitsThreshold = null;
+    private final List<String> objectFormat = new ArrayList<>();
+    private Integer start = null;
+    private Integer stepValue = null;
+    private List<UserDefinedRanking> userDefinedRanking = null;
+    private List<String> sort = null;
+    private List<UserDefinedBoost> userDefinedBoost = null;
+    private Boolean queryDebug = null;
```

Comment:
Der er ikke lavet getter's på alle private members - Er det med vilje?
F.eks. mangler der getters for userDefinedRanking og userDefinedBoost